### ISSUE TITLE
Triage the 150+ Issues

### ISSUE THREAD

**Issue by bms63** [pharmaverse/admiral#805]
Go through the Issues and close ones that are no longer needed, redundant, etc.

Query authors of issue if unclear on what to do.

I'll probably approach it as I did with the massive pruning of branches.  :)

It will be fun!

**Comment by rossfarrugia** [comment]
not all heroes wear capes @bms63

**Comment by bms63** [comment]
Hey all,

Fun times with JSON files!  I managed to extract all the issues in one swoop and compile them into an excel file.

I put it on [Box ](https://exbpbox.box.com/s/tql3br2syswg19f84cyp9twjrc2icdev).  I'm happy to bottom line this and work through each one.   I was thinking that I would insert in the **_Resolution Column_** either **Keep Open** or **Closed** and a short rationale in the **_Notes Column_** on why we are still keeping it open.  

![image](https://user-images.githubusercontent.com/10111024/149234240-f974f470-c5e2-4202-9502-fb06eeb3bc84.png)

This is way as I work through these I don't have to wade through the Issues to figure out if I addressed it.  

Thinking ahead when we have 100+ issues to wade through again...

Where should I put code like this in case we need it again?  It queries the github api and pulls down all the issues.  Pulls out the relevant information from the json list things, cleans it up and send it out to an excel file.  Not super pretty code but it works for a quick and dirty 3 hour project!

Felt like I was in the matrix for minute there!

**Comment by bundfussr** [comment]
I wonder if the triage should be triggered by the total number of issues or if we should base it on the time the issue was not updated or a fixed time after a review is required.

Maybe we could add a "Next Review Date" field to all issues. When the issue is created we could set it to creation date plus x months. In each sprint we could review all issues with "Next Review Date" within the sprint. For all reviewed issues which are not closed the "Next Review Date" is updated. I would expect that this reduces the number of issues to be reviewed considerably.

What do you think?

**Comment by bundfussr** [comment]
I wonder if we could add the "Next Review Date" field directly to the github issue. @duniek , is this possible using the beta project functionality?

I would like to avoid that we have another issue list in a JSON or excel file because it requires additional work to keep them in sync.

For the "Next Review Date" I would propose to set it to three months after issue creation data or last review date.

I am still not sure if the triage should be triggered by the number of issues. If admiral takes off and we get a lot of contributions, the number of issues will increase. However it should not be a problem as long as somebody is working on the issues. The problematic ones are those nobody is working on because they were forgotten or are no longer relevant.

**Comment by bms63** [comment]
In case we ever need to get a flat csv file of all the issues again - I will keep my program here https://github.com/bms63/query_github_issues/blob/main/get_issues.R

**Comment by bms63** [comment]
@bundfussr Agreed that triage should **NOT** be triggered by the number of issues, but triage should be triggered by the length of time the issue has been in our tracker.   This doesn't mean its gets removed just some sort of review is done on it periodically.  Not sure how this can be done, but I'm sure we will figure it out.

Very nervous for when we have more contributors working on this project and our issues really start balloning!!

 I will try and be more active in the product backlog grooming meetings, but I won't be at the one on Monday, January 17th.  Holiday in US.

**Comment by rossfarrugia** [comment]
_"Very nervous for when we have more contributors working on this project and our issues really start balloning!!"_ - hopefully at that stage they'll contribute code/fixes themselves, not only issues!

**Comment by bms63** [comment]
> _"Very nervous for when we have more contributors working on this project and our issues really start balloning!!"_ - hopefully at that stage they'll contribute code/fixes themselves, not only issues!

Haha!  I feel like the more cooks in the kitchen the more issues are going to get opened!  Just want to make sure we properly triage and prioritize them.  We have issues that go back to April 2021 now.  However, when I look at dplyr issues they have one that goes back to 2016 that is actively being worked on! :0 
![image](https://user-images.githubusercontent.com/10111024/149847589-5cd71e80-efbe-47b3-8fb0-605a8d74c7e2.png)

**Comment by bms63** [comment]
@yangc71

**Comment by bms63** [comment]
Hey all,

I went through most of the issues up until the late Fall 2021.  I thought I closed a bunch, but looks like I only managed to **close 8 issues** during all this hoopla!   

A lot of the **Spring 2021-Fall 2021** Issues are pending on the author providing more clarification or the author needs to close it themselves.  It is hard for me to make a call as I was not around during that time period.  Several will get moved to the metacore issues tracker as well.

I see a couple of ACTIONS coming out of this process:

1.   Create a process to Triage Issues during the Backlog Refinement Meeting (BRM)
2.   Tag Old Issues with **Reviewed at BRM on YYYY-MMM-DD** so we at least know they have been looked at and decided they should be kept as Issues
3.  Create a series of **janitor** tasks for keeping admiral Repository clean and healthy.  
      a)  Triage and Clean Branches
      b)  Triage and Clean Pull Requests
      c)  Triage Issues 
 
Perhaps the janitor tasks is something we could prioritize at the Backlog Refinement Meeting going forward?  

Looking through my calendar, **_I don't see any Backlog Refinement Meetings for Admiral_**.  @duniek @thomas-neitmann can I get invited to them or can we start having them again for 2022?  

Ben
